cpp/src/arrow/util/arrow_compression.cc

```diff
@@ -0,0 +1,90 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "arrow/util/compression.h"
```

Comment:
It's not clear from the file names what one would expect to find in "compression.cc" versus "arrow_compression.cc". Can you instead separate each compressor into its own compilation unit with appropriate names:

```
compression.h
compression_zlib.cc
compression_snappy.cc
compression_brotli.cc
compression_zstd.cc
compression_lz4.cc
```

This will also provide more granular control in the build system